stubs/PyAutoGUI/pyautogui/__init__.pyi

```diff
@@ -0,0 +1,230 @@
+from collections.abc import Callable, Generator, Sequence
+from datetime import datetime
+from typing import NamedTuple, TypeVar, overload
+from typing_extensions import ParamSpec
+
+# from pyscreeze import Box
+
+class PyAutoGUIException(Exception): ...
+class FailSafeException(PyAutoGUIException): ...
+class ImageNotFoundException(PyAutoGUIException): ...
+
+_P = ParamSpec("_P")
+_R = TypeVar("_R")
+
+# TODO: Complete types with pyscreeze once we can import it as a type dependency
+# Actually `pyscreeze.Box`, but typeshed doesn't currently have stubs for pyscreeze
+# (and the library doesn't have type annotations either)
+
+class _Box(NamedTuple):
+    left: int
+    top: int
+    width: int
+    height: int
+
+def raisePyAutoGUIImageNotFoundException(wrappedFunction: Callable[_P, _R]) -> Callable[_P, _R]: ...
+def locate(*args, **kwargs) -> _Box | None: ...
+def locateAll(*args, **kwargs) -> Generator[_Box, None, None]: ...
+def locateAllOnScreen(*args, **kwargs) -> Generator[_Box, None, None]: ...
+def locateCenterOnScreen(*args, **kwargs) -> Point | None: ...
+def locateOnScreen(*args, **kwargs) -> _Box | None: ...
+def locateOnWindow(*args, **kwargs) -> _Box | None: ...
+def mouseInfo() -> None: ...
+def useImageNotFoundException(value: bool | None = ...) -> None: ...
+
+KEY_NAMES: list[str]
+KEYBOARD_KEYS: list[str]
+LEFT: str
+MIDDLE: str
+RIGHT: str
+PRIMARY: str
+SECONDARY: str
+QWERTY: str
+QWERTZ: str
+
+def isShiftCharacter(character: str) -> bool: ...
+
+MINIMUM_DURATION: float
+MINIMUM_SLEEP: float
+PAUSE: float
+DARWIN_CATCH_UP_TIME: float
+FAILSAFE: bool
+FAILSAFE_POINTS: list[tuple[int, int]]
+LOG_SCREENSHOTS: bool
+LOG_SCREENSHOTS_LIMIT: int
+G_LOG_SCREENSHOTS_FILENAMES: list[str]
+
+class Point(NamedTuple):
+    x: float
+    y: float
+
+class Size(NamedTuple):
+    width: int
+    height: int
+
+def getPointOnLine(x1: float, y1: float, x2: float, y2: float, n: float) -> tuple[float, float]: ...
+def linear(n: float) -> float: ...
+def position(x: int | None = ..., y: int | None = ...) -> Point: ...
+def size() -> Size: ...
+@overload
+def onScreen(point: tuple[float, float], _: None = ...) -> bool: ...
+@overload
+def onScreen(x: float, y: float) -> bool: ...
+def mouseDown(
+    x: float | Sequence[int] | None = ...,
```

Comment:
I think we can be slightly more permissive here; `int()` is called on all elements of the sequence in `_normalizeXYArgs()` if a sequence is passed, so we can allow `Sequence[float]` instead of `Sequence[int]`. The docstring states that `x` can also be a `str`, as well: https://github.com/asweigart/pyautogui/blob/667161b072e89b4a25b806c5f2ca616e6c0aeec3/pyautogui/__init__.py#L895.

```suggestion
    x: float | str | Sequence[float] | None = ...,
```

Reply:
Good catch, I totally missed that x could be a string >.<